src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/SourceGenConverter.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Text.Json.Serialization.Metadata;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    /// <summary>
+    /// Provides a mechanism to invoke "fast-path" serialization logic via
+    /// <see cref="JsonTypeInfo{T}.Serialize"/>. This type holds an optional
+    /// reference to an actual <see cref="JsonConverter{T}"/> for the type
+    /// <typeparamref name="T"/>, to provide a fallback when the fast path cannot be used.
+    /// </summary>
+    /// <typeparam name="T">The type to converter</typeparam>
+    internal class SourceGenConverter<T> : JsonResumableConverter<T>
```

Comment:
`sealed` on all new internal types. (both in runtime and in the source generator)